Upgrade typescript & shared eslint configuration

NSeydoux opened the pull request:
This upgrades to the latest version of typescript, and fixes the mismatch in the fetch function signature it involves.

NSeydoux committed:
Adapt fetch signature to latest lib.dom.ts

matthieubosquet reviewed (approved):
Updating TypeScript is fine, but you mixed it up with https://github.com/inrupt/solid-client-access-grants-js/pull/277/files.

Maybe remove the issueAccess* changes before merging?

NSeydoux commented:
Yeah #277 was required to pass e2e tests. I was planing on merging it first, and then rebase this on top.

ThisIsMissEm commented on package.json:
```diff
@@ -81,8 +81,8 @@
   "devDependencies": {
     "@inrupt/eslint-config-base": "^1.0.0",
     "@inrupt/eslint-config-lib": "^1.0.0",
```
```suggestion
    "@inrupt/eslint-config-lib": "^1.0.0",
```

You no longer need `base` here.

ThisIsMissEm replied:
Actually, ignore that. This repo hasn't been correctly upgraded to 1.0.0

ThisIsMissEm reviewed (changes requested):
@NSeydoux I'm going to take this over, as we need to do the correct upgrade to the new eslint configs, so I'll push a commit with the changes required for that.

ThisIsMissEm commented:
Okay, the Vercel crash here is related to the sourceurl plugin and typedoc 0.23.3, due to this breaking change:
> Removed ProjectReflection.files, this was an internal cache that should not have been exposed, and shouldn't have existed in the first place, since removing it made TypeDoc faster.

I'm going to remove the sourceurl plugin for now, as it was a "nice to have" on Vercel, but worked out of the box on the published docs.

ThisIsMissEm committed:
chore: remove sourceurl plugin

This plugin gave us source urls on Vercel, but isn't compatible with the latest typedoc, so for now we're removing it

ThisIsMissEm committed:
chore: upgrade typescript and shared eslint configuration

ThisIsMissEm committed:
chore: upgrade playwright

ThisIsMissEm committed:
chore: remove unneeded development dependencies

ThisIsMissEm committed:
chore: upgrade solid-client-authn to 1.12.1

ThisIsMissEm committed:
chore: remove unneeded eslint rules

ThisIsMissEm committed:
chore: add package-lock.json for e2e browser testApp

ThisIsMissEm committed:
ci: install subproject dependencies to prevent linting warnings

ThisIsMissEm reviewed (approved):
I think this is now good to go.

ThisIsMissEm committed:
chore: improve strictness of the engines supported

This explicitly disables Node 18, as we don't yet support it

ThisIsMissEm committed:
chore: remove glob-parent override
